fields/descriptor.go

```diff
@@ -54,7 +54,7 @@ func (d Descriptor) ValidateSQLType() error {
 	return d.Scanner().Scan(val)
 }
 
-func (d Descriptor) copy(from, to reflect.Value, isValid bool) {
+func (d *Descriptor) copy(from, to reflect.Value, isValid bool) {
```

Comment:
My takeaway here is: everything that's passed into a function, including pointers, is pass by value in go. If we have `Descriptor`, we are copying a value (all struct fields) and then pointing to it. If we have `*Descriptor`, we are copying a pointer (just one thing pointing to the value). The second is lighter because we don't have to re-allocate the struct and its children, we just have to allocate the pointer.